espnet/nets/pytorch_backend/transducer/loss.py

```diff
@@ -2,29 +2,47 @@
 
 """Transducer loss module."""
 
-from torch import nn
+import logging
+import sys
+import torch
 
-from warprnnt_pytorch import RNNTLoss
 
-
-class TransLoss(nn.Module):
-    """Transducer loss.
+class TransLoss(torch.nn.Module):
+    """Transducer loss module.
 
     Args:
         trans_type (str): type of transducer implementation to calculate loss.
         blank_id (int): blank symbol id
-
     """
 
     def __init__(self, trans_type, blank_id):
         """Construct an TransLoss object."""
         super(TransLoss, self).__init__()
 
+        device = torch.device("cuda" if torch.cuda.is_available() else "cpu")
+
         if trans_type == "warp-transducer":
+            from warprnnt_pytorch import RNNTLoss
+
             self.trans_loss = RNNTLoss(blank=blank_id)
+        elif trans_type == "warp-rnnt":
+            if device.type == "cuda":
+                try:
+                    from warp_rnnt import rnnt_loss
+
+                    self.trans_loss = rnnt_loss
+                except ImportError:
+                    logging.error(
+                        "warp-rnnt is not installed. Please re-setup"
+                        " espnet or use 'warp-transducer'"
+                    )
+                    sys.exit(1)
```

Comment:
Oh, it's not allowed to use sys.exit.
Please raise the error as it is.

Reply:
Oh sorry, I wanted to exit without stopping the process (`exit(1)`) because warp-rnnt install is optional. What would be the best option in that case?

Reply:
Or, raise RuntimeError with the message.

Reply:
Maybe, you misunderstood. sys.exit(1) is almost equivalent to raise SystemExit, so it will stop the process.

I'm still not sure your intension. Do you mean it should be fallback to warp-transducer?

Reply:
Oh, sorry misunderstood what part it was, will change that!

Reply:
Oh my, I understand why I did that: I was tired and lazy when adding test for warp-rnnt and used a sys.exit here to avoid a failure with pytest...

Reply:
Hmm?

Reply:
Don't mind, sorry, I changed that!